Basic/ClientDriven/Assets/Scripts/NetworkObjectSpawner.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using Unity.Netcode;
+using UnityEngine;
+using UnityEngine.Networking;
+using Random = System.Random;
+
+namespace Unity.Multiplayer.Samples.ClientDriven
+{
+    public class NetworkObjectSpawner : MonoBehaviour
+    {
+        [SerializeField] NetworkObject prefabReference;
+
+        public void Start()
+        {
+            if (NetworkManager.Singleton != null)
+            {
+                NetworkManager.Singleton.OnServerStarted += OnServerStartedIngredientSpawn;
+            }
+        }
+
+        void OnDestroy()
+        {
+            if (NetworkManager.Singleton != null)
+            {
+                NetworkManager.Singleton.OnServerStarted -= OnServerStartedIngredientSpawn;
+            }
+        }
+
+        void OnServerStartedIngredientSpawn()
+        {
+            Random randomGenerator = new Random();
+            var instantiatedNetworkObject = Instantiate(prefabReference, transform.position, transform.rotation, null);
```

Comment:
nitpick: "var"'s type is unclear on this line. It's recommended to use var only if the code you're writing is explicit about the type you're working on (I.E: `var randomGenerator = new Random();` is fine)

Reply:
There is another spawner script in the same project which also uses a var for this line. What could I use instead of var here? (GameObject?)

Reply:
You should use the type that the Instantiate is returning (if you hover it, it wil ltell you).
Please do not change other scripts, a "var"-related refactoring is not in the scope of this PR.

Reply:
I did that in the last change and no worries I will not touch other scripts :)